Java/TelemetrySystem/src/main/java/tddmicroexercises/telemetrysystem/services/TelemetryDiagnosticControls.java

```diff
@@ -0,0 +1,40 @@
+package tddmicroexercises.telemetrysystem.services;
+
+import tddmicroexercises.telemetrysystem.interfaces.DiagnosticControls;
+
+public class TelemetryDiagnosticControls implements DiagnosticControls {
+    private final String DiagnosticChannelConnectionString = "*111#";
+    private final ClientService clientService;
+    private final ConnectionService connectionService;
+    private String diagnosticInfo = "";
+
+    public String getDiagnosticInfo() {
+        return diagnosticInfo;
+    }
+
+    public TelemetryDiagnosticControls(ClientService clientService, ConnectionService connectionService) {
+        this.clientService = clientService;
```

Comment:
To ensure dependency inversion principle we must pass the type as interface name instead of actual class name (e.g the type of clientService should be Client interface as it is implemented by ClientService class)

Reply:
Sure , will change